ui/page/components/consensus_list.go

```diff
@@ -0,0 +1,145 @@
+		layout.Rigid(layoutAgendaDetails(l, "Voting Preference: "+agenda.VotingPreference)),
+		layout.Rigid(func(gtx C) D {
+			return layoutAgendaVoteAction(gtx, l, consensusItem)
+		}),
+	)
+}
+
+func layoutAgendaStatus(gtx C, l *load.Load, agenda dcrlibwallet.Agenda) D {
+
+	var statusLabel decredmaterial.Label
+	var statusIcon *decredmaterial.Icon
+	var backgroundColor color.NRGBA
+
+	switch agenda.Status() {
+	case "Finished":
+		statusLabel = l.Theme.Label(values.MarginPadding14, agenda.Status())
+		statusLabel.Color = l.Theme.Color.GreenText
+		statusIcon = decredmaterial.NewIcon(l.Icons.NavigationCheck)
+		statusIcon.Color = l.Theme.Color.Green500
+		backgroundColor = l.Theme.Color.Green50
+		canVote = false
+	case "In progress":
+		clr := l.Theme.Color.Primary
+		statusLabel = l.Theme.Label(values.MarginPadding14, agenda.Status())
+		statusLabel.Color = clr
+		statusIcon = decredmaterial.NewIcon(l.Icons.NavMoreIcon)
+		statusIcon.Color = clr
+		backgroundColor = l.Theme.Color.LightBlue
+		canVote = true
+	case "Upcoming":
+		statusLabel = l.Theme.Label(values.MarginPadding14, agenda.Status())
+		statusLabel.Color = l.Theme.Color.Text
+		statusIcon = decredmaterial.NewIcon(l.Icons.PlayIcon)
+		statusIcon.Color = l.Theme.Color.DeepBlue
+		backgroundColor = l.Theme.Color.Gray2
+		canVote = false
+	}
+
+	return layout.Flex{Spacing: layout.SpaceBetween}.Layout(gtx,
+		layout.Rigid(func(gtx C) D {
+			lbl := l.Theme.Label(values.MarginPadding20, (strings.Title(strings.ToLower(agenda.AgendaID))))
+			lbl.Font.Weight = text.SemiBold
+			return layout.Flex{}.Layout(gtx,
+				layout.Rigid(lbl.Layout),
+			)
+		}),
+		layout.Rigid(func(gtx C) D {
+			return decredmaterial.LinearLayout{
+				Background: backgroundColor,
+				Width:      decredmaterial.WrapContent,
+				Height:     decredmaterial.WrapContent,
+				Direction:  layout.Center,
+				Alignment:  layout.Middle,
+				Border:     decredmaterial.Border{Color: backgroundColor, Width: values.MarginPadding1, Radius: decredmaterial.Radius(10)},
+				Padding:    layout.Inset{Top: values.MarginPadding3, Bottom: values.MarginPadding3, Left: values.MarginPadding8, Right: values.MarginPadding8},
+				Margin:     layout.Inset{Left: values.MarginPadding10},
+			}.Layout(gtx,
+				layout.Rigid(func(gtx C) D {
+					return layout.Inset{Right: values.MarginPadding4}.Layout(gtx, func(gtx C) D {
+						return statusIcon.Layout(gtx, values.MarginPadding16)
+					})
+				}),
+				layout.Rigid(statusLabel.Layout))
+		}),
+	)
+}
+
+func layoutAgendaDetails(l *load.Load, data string) layout.Widget {
+	return func(gtx C) D {
+		lbl := l.Theme.Label(values.MarginPadding16, data)
+		lbl.Font.Weight = text.Light
+		return layout.Inset{Top: values.MarginPadding10}.Layout(gtx, lbl.Layout)
+	}
+}
+
+func layoutAgendaVoteAction(gtx C, l *load.Load, item *ConsensusItem) D {
+	gtx.Constraints.Min.X, gtx.Constraints.Max.X = gtx.Px(unit.Dp(150)), gtx.Px(unit.Dp(150))
+	if canVote {
```

Comment:
You don't need a public canVote property that is set in one function and used in a different function.
```suggestion
	if item.Agenda.Status() == "In progress" { // remember to use a dcrlibwallet constant instead
```